src/components/e-icon.vue

```diff
@@ -30,226 +49,82 @@
-          }
-        }
-
-        svg.setAttribute('tabindex', this.tabindex);
-        svg.setAttribute('role', 'img');
-        svg.setAttribute('aria-label', this.icon);
-        svg.setAttribute('focusable', this.focusable);
-      },
-
-      /**
-       * Creates a new SVG element if not already in cache.
-       *
-       * @param {String} content - The SVG content as string.
-       *
-       * @returns {Node}
-       */
-      createSvgElement(content) {
-        if (!nodeCache[this.icon]) {
-          const container = document.createElement('div');
-
-          container.innerHTML = content;
-
-          nodeCache[this.icon] = container;
-        }
-
-        return nodeCache[this.icon].cloneNode(true).children[0];
-      },
-
+    computed: {
       /**
-       * Gets an SVG element for the given content.
-       *
-       * @param {String} content - The SVG content as string.
+       * Returns a viewBox definition object.
        *
-       * @returns {Node}
-       */
-      getSvgElement(content) {
-        const svg = this.createSvgElement(content);
-
-        this.setAttributes(svg);
-
-        return svg;
-      },
-
-      /**
-       * Load the icon.
+       * @returns {Object}
        */
-      loadIcon() {
-        if (!this.inline || !this.src) {
-          return;
-        }
-
-        if (!cache[this.icon]) {
-          cache[this.icon] = this.$axios
-            .get(this.src)
-            .then(response => response.data);
+      viewBox() {
+        const { icon } = this;
+        const lookup = sizeLookup[icon];
+        const size = this.size?.split(' ') || [defaultSize];
+
+        // Auto map height for non square icons.
+        if (size.length === 1 && lookup) {
+          size[1] = (size[0] / lookup[0]) * lookup[1];
         }
 
-        cache[this.icon].then((svg) => {
-          const { $el } = this;
-
-          // remove previous icon
-          while ($el.firstChild) {
-            $el.removeChild($el.firstChild);
-          }
-
-          $el.appendChild(this.getSvgElement(svg));
-        });
+        return {
+          width: size[0],
+          height: size[1] || size[0],
+        };
       },
     },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    // methods: {},
+    // render() {},
   };
 </script>
 
 <style lang="scss">
   .e-icon {
-    display: inline-block;
-
-    svg {
-      display: block; // Removes additional letter spacing around element.
-      pointer-events: none; // Prevents IE11 from swallowing events.
-    }
-
-    &--color-gray svg path {
-      fill: $color-grayscale--400;
-    }
-
-    &--color-lightgray svg path {
-      fill: $color-grayscale--600;
-    }
-
-    &--color-white svg path {
-      fill: $color-grayscale--1000;
-    }
+    display: block;
```

Comment:
Should we use `display: inline-block` to be able to use the icon within an inline element?

Reply:
@janwidmer I would not recommend to use `inline-block` as default, because it adds under length to the element. Additionally we mostly use flex to align text and icons anyway, so it would not be of big use. OK?

Reply:
I usually used line-height of 0 to prevent under lenght.. but it's fine like this for me.